### PR TITLE
Crud attributes

### PR DESCRIPTION
**Pull request by mduc28**
Anh check cho em phần model của 2 cái attributes. Em muốn thị cái phần attribute type trong view list của cái Attribute Value thì nó báo lỗi "Trying to get property 'name' of non-object". Em có dd để check thì n gửi được cái relation nhưng lại không được.

### CODE REVIEW

File: app/Http/Controllers/admin/AttributeValuesController.php [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+     */
+    public function store(Request $request)
+    {
+        $validator = Validator::make(
+            $request->all(),
+            [
+                'name' => 'max:255|min:3'
+            ],
+            [
+                'max' => 'Name is too long!',
```
Comment (hieund97) [inline app/Http/Controllers/admin/AttributeValuesController.php]: show message more clearer and specific condition. Ex: Name is too long ( max: 255 character)

File: resources/views/admin/attributes/value/list.blade.php [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+                                        <tr>
+                                            <th width='20%'>ID</th>
+                                            <th width='40%'>Name</th>
+                                            <th width='30%'>Attribute Type</th>
+                                            <th width='20%'>Action</th>
+                                        </tr>
+                                    </thead>
+                                    <tbody>
+                                        @foreach ($aryAttributeValues as $value)
+                                            {{-- @dd($value) --}}
```
Comment (hieund97) [inline resources/views/admin/attributes/value/list.blade.php]: remove

File: resources/views/admin/attributes/value/list.blade.php [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+                                            <tr>
+                                                <td>{{ $value->id }}</td>
+                                                <td>{{ $value->name }}</td>
+                                                <td>{{--{{ $value->attributes->name }}--}}</td>
```
Comment (hieund97) [inline resources/views/admin/attributes/value/list.blade.php]: remove

File: app/Models/AttributeTypes.php [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+    protected $fillable = [
+        'name', 
+    ];
+
+    /**
+     * Get all of the attributesType for the AttributeTypes
+     *
+     * @return \Illuminate\Database\Eloquent\Relations\HasMany
+     */
+    public function attributes()
```
Comment (hieund97) [inline app/Models/AttributeTypes.php]: 2 function can't have same name. When you try to access relationship via this function, laravel can't detect what relationship do you want to access.

File: app/Http/Controllers/admin/AttributeValuesController.php [hunk 5]
```diff
@@ -0,0 +1,116 @@
...
+     */
+    public function store(Request $request)
+    {
+        $validator = Validator::make(
+            $request->all(),
+            [
+                'name' => 'max:255|min:3'
+            ],
+            [
+                'max' => 'Name is too long! (max: 255 characters)',
```
Comment (hieund97) [inline app/Http/Controllers/admin/AttributeValuesController.php:39]: Use `:attribute` in message and apply for all in this PR

File: app/Http/Controllers/admin/AttributeValuesController.php [hunk 6]
```diff
@@ -0,0 +1,116 @@
...
+    public function update(Request $request, $id)
+    {
+        $validator = Validator::make(
+            $request->all(),
+            [
+                'name' => 'max:255|min:3'
+            ],
+            [
+                'max' => 'Name is too long! (max: 255 characters)',
+                'min' => 'Name is too short!'
```
Comment (hieund97) [inline app/Http/Controllers/admin/AttributeValuesController.php]: You changed message for `max` rule. What about `min` rule?

**Comment by hieund97** [comment]
@mduc28 check comment

File: app/Models/AttributeTypes.php (hunk 4 above)
Comment (mduc28) [reply app/Models/AttributeTypes.php]: But in models of blogs and categories, I still can use their functions when both functions' name is categories.

Comment (mduc28) [reply app/Models/AttributeTypes.php]: I have tried to change their names but it still doesn't work

**Pull request merged**